### PR TITLE
chore: add injective-testing package 

### PR DESCRIPTION
**Pull request by Tofel**
supports  testing multiple smart contract testing

### CODE REVIEW

**Review by AntekM — APPROVED** [review]
Look ok, I would add some short description/example how to use it, and especially mention all issues that we have found so far (like all contracts involved need to has same in/out types)

**Comment by Tofel** [comment]
I added comment here: https://github.com/Ninja-Finance/ninja-contracts/pull/110

File: packages/injective-testing/src/chain_mock.rs [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+        }
+
+        self.state.execs.borrow_mut().push(msg);
+        exec_calls_count += 1;
+
+        if self.responses.executes.is_empty()
+            || exec_calls_count > self.responses.executes.len()
+            || self.responses.executes[exec_calls_count - 1].is_empty()
+        {
+            Ok(AppResponse::default())
```
Comment (gorgos) [inline packages/injective-testing/src/chain_mock.rs]: I know people do this often in Rust, but I think it's so much more readable to here instead do

```
    return Ok(AppResponse::default());
}

[normal code without else]
```

File: packages/injective-testing/src/chain_mock.rs [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        } else {
+            let stored_result = self.responses.executes.get(exec_calls_count - 1).unwrap().response.as_ref().unwrap();
+
+            //In order to implement the trait that method has to receive &self and neither Result nor Binary implements Copy
+            //and that the reason why I'm manually copying the underlying [u8] in order to return owned data
```
Comment (gorgos) [inline packages/injective-testing/src/chain_mock.rs]:
```suggestion
            // In order to implement the trait that method has to receive &self and neither Result nor Binary implements Copy
            // and that's the reason why I'm manually copying the underlying [u8] in order to return owned data
```

File: packages/injective-testing/src/chain_mock.rs [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        if self.responses.queries.is_empty()
+            || query_calls_count > self.responses.queries.len()
+            || self.responses.queries[query_calls_count - 1].is_empty()
+        {
+            Ok(Binary::default())
+        } else {
+            let stored_result = self.responses.queries.get(query_calls_count - 1).unwrap().response.as_ref().unwrap();
+
+            //In order to implement the trait that method has to receive &self and neither Result nor Binary implements Copy
+            //and that the reason why I'm manually copying the underlying [u8] in order to return owned data
```
Comment (gorgos) [inline packages/injective-testing/src/chain_mock.rs]:
```suggestion
            // In order to implement the trait that method has to receive &self and neither Result nor Binary implements Copy
            // and that's the reason why I'm manually copying the underlying [u8] in order to return owned data
```

File: packages/injective-testing/src/chain_mock.rs [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+
+    fn sudo<ExecC, QueryC>(
+        &self,
+        _api: &dyn Api,
+        _storage: &mut dyn Storage,
+        _router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        _block: &BlockInfo,
+        msg: Self::SudoT,
+    ) -> AnyResult<AppResponse> {
+        bail!("Unexpected sudo msg {:?}", msg)
```
Comment (gorgos) [inline packages/injective-testing/src/chain_mock.rs:274]: Nice, new macro learnt.

**Review by gorgos — CHANGES_REQUESTED** [review]
Nice additions, look good to me, just minor comments.

File: packages/injective-testing/src/chain_mock.rs (hunk 4 above)
Comment (Tofel) [reply packages/injective-testing/src/chain_mock.rs:274]: 👍

File: packages/injective-testing/src/chain_mock.rs (hunk 1 above)
Comment (Tofel) [reply packages/injective-testing/src/chain_mock.rs]: yeah, but I'll stick to Rust way ;-)

Comment (gorgos) [reply packages/injective-testing/src/chain_mock.rs]: Why? I don't think we should follow bad patterns.

**Review by gorgos — APPROVED** [review]

**Pull request merged**